infra/modules/azure_app_service_plan_autoscaler/variables.tf

```diff
@@ -0,0 +1,157 @@
+variable "tags" {
+  type        = map(any)
+  description = "Resources tags"
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "Resource group to deploy resources to"
+}
+
+variable "target_service" {
+  type = object({
+    app_service_name  = optional(string)
+    function_app_name = optional(string)
+  })
+
+  validation {
+    condition = (
+      (var.target_service.app_service_name != null && var.target_service.function_app_name == null) ||
+      (var.target_service.app_service_name == null && var.target_service.function_app_name != null)
+    )
+    error_message = "Only one between \"app_service_name\" and \"function_app_name\" can have a value. It is not possible to set both of them \"null\"."
+  }
+}
+
+variable "scheduler" {
+  type = object({
+    high_load = optional(object({
+      start = object({
+        hour    = number
+        minutes = number
+      })
+      end = object({
+        hour    = number
+        minutes = number
+      })
+      name    = string
+      default = number
+      minimum = number
+    }), null)
+    low_load = optional(object({
+      start = object({
+        hour    = number
+        minutes = number
+      })
+      end = object({
+        hour    = number
+        minutes = number
+      })
+      name    = string
+      default = number
+      minimum = number
+    }), null)
+    normal_load = object({
+      default = number
+      minimum = number
+    })
+    maximum = optional(number, 30)
+  })
+
+  default = {
+    high_load = {
+      name = "high_load_profile"
+      start = {
+        hour    = 19
+        minutes = 30
+      }
+      end = {
+        hour    = 22
+        minutes = 59
+      }
+      default = 12
+      minimum = 4
+    }
+    low_load = {
+      name = "low_load_profile"
+      start = {
+        hour    = 23
+        minutes = 00
+      }
+      end = {
+        hour    = 05
+        minutes = 00
+      }
+      default = 10
+      minimum = 2
+    }
+    normal_load = {
+      default = 11
+      minimum = 3
+    }
+    maximum = 30
+  }
+
+  description = "Set the recurrent autoscaling profiles, including start and end time ([hh]:[mm]), the minimum and maximum number of instances and the default value (used when metrics are not available for some technical issue). Each default value can be overridden"
```

Comment:
I'd rename default to `fallback`. Moreover, can we make it clear how the specified interval works against the min/max values? it's not clear what happens outside / inside  that interval